modules/globallock/lock.go

```diff
@@ -57,11 +58,15 @@ func newMemoryLockService() *memoryLockService {
 	}
 }
 
-func (l *memoryLockService) GetLock(name string) Locker {
+func (l *memoryLockService) GetLocker(name string) Locker {
 	v, _ := l.syncMap.LoadOrStore(name, &memoryLock{})
 	return v.(*memoryLock)
 }
 
+func (l *memoryLockService) RemoveLocker(name string) {
+	l.syncMap.Delete(name)
```

Comment:
Wrong.

----

A: GetLocker() -> L1
A: L1.Lock()
A: run ...

B: GetLocker() -> L1
B: L1.Lock() // wait

A: L1.Unlock()
B: run ...
A: RemoveLocker() // L1 is removed
C: GetLocker() -> L2 // gets a new one
C: L2.Lock()
C: run ....

Now, B and C both are running on the same "locker name"

Reply:
The removelocker should be invoked after the locker haven't been used. i.e. After pull request closed/merged, no code will get locker. To reduce the pool key/value, then we can invoke RemoveLocker.

Reply:
I know what you are doing, but it is wrong.